ls1intum/Artemis
# `Development`: Display LTI login name for newly generated Artemis users

> [!WARNING]  
> **Deploy only locally or on ts11!** The LTI feature is only available on TS11.

### Checklist
#### General


#### Server

#### Client

### Motivation and Context


Fixes #7483 

### Description

Login name added for generated password pop-up. The predefined login name is filled in while accessing Artemis via an external LMS (e.g. Moodle).

### Steps for Testing

Prerequisites:
-  Please use TS11 to test this PR since lti profile is enabled
- 1 Student

1. Navigate to https://moodle.ase.in.tum.de/
2. Login using one of the provided users below. Please check the used check box so other testers won't test with the same user.
3. Navigate to "LTI Testing Course"
4. Select one of the exercises e.g. "TS11 Authentication Test Exercise"
5. Verify that a pop-up shows up with login and password credentials.
6. Copy the shown username and password to a notepad.
7. Check the acknowledge checkbox and confirm to close the pop-up.
8. Log out from Artemis inside the iframe in Moodle.
9. Reload the page and verify you see the login page.
10. Verify username is filled and the same as the username from the 7th step.

List of sample users for moodle: (users are ready for re-testing again after final fixes)
- [x] username:  jane_doe password: use same password for artemis_test_user_1 
- [x] username:  joey_read password: use same password for artemis_test_user_1
- [x] username:  barney_young password: use same password for artemis_test_user_1 
- [x] username:  john_smith password: use same password for artemis_test_user_1 

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage







#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [lti13-exercise-launch.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Client-Tests/app/lti/lti13-exercise-launch.component.ts.html) | 100% | ✅ |
| [lti-initializer-modal.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/exercise-details/lti-initializer-modal.component.ts.html) | 78.57% | ✅ |
| [lti-initializer.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/exercise-details/lti-initializer.component.ts.html) | 100% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [Lti13LaunchFilter.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.security.lti/Lti13LaunchFilter.html) | 86% | ✅ |
| [Lti13Service.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.lti/Lti13Service.html) | 93% | ✅ |



### Screenshots



Login name information on pop-up:

![image](https://github.com/ls1intum/Artemis/assets/11020241/4a0ea1b4-10f8-4313-ba6f-002641737c8a)

Prefilled username:

![image](https://github.com/ls1intum/Artemis/assets/11020241/2f04010d-d758-4a3f-9dea-e6694fc4c2fa)
